modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscription.java

```diff
@@ -248,12 +248,14 @@ public void onMessageAck(long messageID) throws AndesException {
      *
      * @param messageID id of the message acknowledged
      * @param reQueue true if message should be re-queued to subscriber
+     * @param isMessageBeyondLastRollback true if message was rejected after last rollback
      * @throws AndesException on a message re-schedule issue
      * @return DeliverableAndesMetadata reference of rejected message
      */
-    public DeliverableAndesMetadata onMessageReject(long messageID, boolean reQueue)
-            throws AndesException {
+    public DeliverableAndesMetadata onMessageReject(long messageID, boolean reQueue,
+                                                    boolean isMessageBeyondLastRollback) throws AndesException {
         DeliverableAndesMetadata rejectedMessage = subscriberConnection.onMessageReject(messageID);
+        rejectedMessage.setIsBeyondLastRollbackedMessage(isMessageBeyondLastRollback);
```

Comment:
Should this be done before subscriberConnection.onMessageReject?

Reply:
What on message reject does is removing the message from outbound tracker and getting the message reference at the server side. We need to call setIsBeyondLastRollbackedMessage on that message.